### PR TITLE
Add tests for update write repair edge cases

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review

This adds explicit edge case testing for the following:
- Replicas are inconsistent with diverging document versions
- Replicas are inconsistent with one replica not having the document

Also disable merges in existing inconsistent update test. This has more obvious semantics than setting max pending ideal state ops to zero.

### CODE REVIEW

File: tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    response = vespa.document_api_v1.http_get(path)
+    vespa.document_api_v1.assert_response_ok(response)
+    JSON.parse(response.body)['fields']
+  end
+
+  def feed_doc_with_field_value(title:)
+    # Also add a second field that updates won't touch so that we can detect if
+    # a 'create: true' update erroneously resets the state on any replica.
+    doc = { 'fields' => { 'title' => title, 'artist' => 'cool dude' } }
+    response = api_http_post("/document/v1/storage_test/music/number/1/foo", doc.to_json)
```
Comment (geirst) [inline tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb]: The DocumentApiV1 class has some helper functions (put, get, remove) that constructs the path for you. Consider using get put() function instead.

File: tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      response)
+    response
+  end
+
+  def update_doc_with_field_value(title:, create_if_missing:)
+    # Use 'create: true' update to ensure that not performing a write repair as
+    # expected will create a document from scratch on the node.
+    update = { 'fields' => { 'title' => { 'assign' => title } } }
+    update['create'] = true if create_if_missing
+    response = api_http_put("/document/v1/storage_test/music/number/1/foo", update.to_json)
```
Comment (geirst) [inline tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb]: Consider adding a helper function for updates on DocumentApiV1.

File: tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  def mark_content_node_down(idx)
+    mark_node_in_state(idx, 'd')
+  end
+
+  def mark_content_node_up(idx)
+    mark_node_in_state(idx, 'u')
+  end
+
+  def verify_document_has_expected_contents(title:)
+    fields = api_http_get('/document/v1/storage_test/music/number/1/foo')
```
Comment (geirst) [inline tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb]: Consider using DocumentApiV1.get() instead.

File: tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb (hunk 1 above)
Comment (vekterli) [reply tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb]: Done

File: tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb (hunk 2 above)
Comment (vekterli) [reply tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb]: Done

File: tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb (hunk 3 above)
Comment (vekterli) [reply tests/search/inconsistent_replicas/updates_to_inconsistent_buckets.rb]: Done

**Comment by vekterli** [comment]
@geirst PTAL at the latest changes
